### DISCUSSION TITLE
'socketio.sgunicorn.GeventSocketIOWorker'

### DISCUSSION THREAD

**Discussion by qasas145** [miguelgrinberg/Flask-SocketIO#1643] [General]
hello miguel
@miguelgrinberg 
i have made django+react+socketio project and i when i try to deploy it to heroku i get this error messages 'socketio.sgunicorn.GeventSocketIOWorker'
this my Procfile file :
`release: python manage.py migrate
web: gunicorn --worker-class socketio.sgunicorn.GeventSocketIOWorker --log-file=- backend.wsgi:application`
and this my wsgi.py

`
import os

from django.core.wsgi import get_wsgi_application
import socketio
import eventlet
import gevent
os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'backend.settings')
application = get_wsgi_application()
sio=socketio.Server(async_mode='gevent',cors_allowed_origins='*', logger=True, engineio_logger=True, ping_timeout=2)
application = socketio.WSGIApp(sio, application)
@sio.on('mohamed')
def mohamed(data, datamain) :
    print('the main s the main of the function that has the most of the ten to use in the game')
    print('this the function that has teh name of the ten to use in te div of text')
    print(datamain)
    sio.emit('res', {
        "name" :"this the response of the await of the function"
    })
eventlet.wsgi.server(eventlet.listen(('', 8000)), application)
`

and this the App.js
`
import React, {useEffect} from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router , Switch, Link, Route} from "react-router-dom";
import Home from './Home'
import axios from 'axios';
import endpoint from "./endpoint"
const App=()=>{
  const getData=async()=>{
    const res=await fetch('http://localhost:8000/api/try/', {
      method :"POST",
      headers :{
        "Content-Type": "application/json"
      },
      body :JSON.stringify({
        email :"mohamed"
      })
    })
    const data=await res.json();
    console.log(data)
  }
  useEffect(()=>{
    getData();
  })
  return (
      <Router>
        <Link to="/login">login</Link>
        <Link to="/home">Home</Link>
          <Switch>
            <Route path="/home" component={Home} />
            {/* <Route path="/login" component={Login}/> */}
          </Switch>
      </Router>
  )
}
export default App;
`

and there is another problem when i try to connect the backend with that api("http://localhost:8000/")  give me API has been blocked by the client ..

**Comment by miguelgrinberg** [comment]
You are confusing this package with another one. See this issue for more details: https://github.com/miguelgrinberg/Flask-SocketIO/issues/294. If you are using this package, then follow the docs referenced in the readme file for your deployment instructions.

**Reply by qasas145** [reply]
@miguelgrinberg 
ok i have put `gunicorn -k gevent -w 1 backend.wsgi:applications`
in Procfile
and worked and give me listening at: http://0.0.0.0:5031
 but there is still a problem  that is when i put in App.js
fetch('http://localhost:8000/') give me in the console
http://localhost:8000/api/try/ net::ERR_BLOCKED_BY_CLIENT

**Reply by miguelgrinberg** [reply]
Did you configure CORS appropriately?

**Reply by qasas145** [reply]
@miguelgrinberg 
ALLOWED_HOSTS = ['*']
yes i put this in settings.py
`CORS_ORIGIN_ALLOW_ALL=True
APPEND_SLASH=False
CORS_ORIGIN_ALLOW_ALL = True
CORS_ALLOW_CREDENTIALS = True`
and this in installed app
`'corsheaders',`
anything else ?
or it can be because the server is listening in other post not 8000?

**Reply by miguelgrinberg** [reply]
That's all for the Django side, you have to configure Socket.IO for CORS, that's separate. Are you reading the documentation?

**Reply by qasas145** [reply]
yes @miguelgrinberg 
that is in wsgi.py
sio=socketio.Server(async_mode='gevent',cors_allowed_origins='*', logger=True, engineio_logger=True, ping_timeout=2) 
but also when i remove the socketio and make the app run give me that messages too

**Reply by miguelgrinberg** [reply]
The only part I can help you with is Socket.IO, and you have done your configuration. I wouldn't set `*` for CORS, but it's not for me to judge. I cannot help you if you have problems elsewhere in your application, you have to debug this yourself as I have no knowledge of your application structure or configuration. I also don't use Django a lot, not my thing.
